src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -411,13 +409,18 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, DiagnosticBag diag
                         var mods = singleDeclaration.Modifiers;
                         if ((mods & DeclarationModifiers.Partial) == 0)
                         {
-                            diagnostics.Add(ErrorCode.ERR_MissingPartial, singleDeclaration.NameLocation, self.Name);
+                            diagnostics.Add(ErrorCode.ERR_MissingPartial, singleDeclaration.NameLocation, this.Name);
                             modifierErrors = true;
                         }
                     }
                 }
             }
 
+            if (this.Name == "record")
```

Comment:
Should this warning only be issued if the selected language version >= IDS_FeatureRecords.RequiredVersion() ?

Reply:
Yes, this needs to be tied to C# 9.

---
In reply to: [476373766](https://github.com/dotnet/roslyn/pull/47094#discussion_r476373766) [](ancestors = 476373766)

Reply:
consider using something like `this.Name == SyntaxFacts.GetText(SyntaxKind.RecordKeyword)` or `SyntaxFacts.GetContextualKeywordKind(this.Name) == SyntaxKind.RecordKeyword`